### PR TITLE
Allow custom space index sort

### PR DESCRIPTION
**Pull request by OAGr**
image

### CODE REVIEW

File: src/components/spaces/index/index.js [hunk 1]
```diff
@@ -16,29 +55,65 @@ function mapStateToProps(state) {
...
     this.props.dispatch(search.fetchNextPage())
   }
+
+  _changeSearchValue(e) {
+    this.setState({searchValue: e.target.value, timeframe: TIMEFRAME_ALL_TIME}, this._fetch)
+  }
+
+  _changeSortBy(sortBy){
+    this.setState({sortBy}, this._fetch)
+
```
Comment (OAGr) [inline src/components/spaces/index/index.js]: Remove newline.

File: src/components/spaces/index/index.js [hunk 2]
```diff
@@ -1,12 +1,51 @@
...
 
 import './style.css'
 
+const TIMEFRAME_ALL_TIME = 'ALL_TIME'
+const TIMEFRAME_MONTHLY = 'MONTHLY'
+const SORT_BY_POPULAR = 'POPULAR'
+const SORT_BY_RECENT = 'RECENT'
+
+const Filters = ({sortBy, timeframe, onChangeSortBy, onChangeTimeFrame}) => {
+  let sortNames = {}
```
Comment (mmcdermott) [inline src/components/spaces/index/index.js]: I would just declare these inline; when I read a 'let' declaration I presume the variable is going to dynamically change later.

File: src/components/spaces/index/index.js [hunk 3]
```diff
@@ -1,12 +1,51 @@
...
+  sortNames[SORT_BY_RECENT] = 'Recent'
+  sortNames[SORT_BY_POPULAR] = 'Popular'
+
+  let timeframeNames = {}
```
Comment (mmcdermott) [inline src/components/spaces/index/index.js]: ditto

File: src/components/spaces/index/index.js [hunk 4]
```diff
@@ -1,12 +1,51 @@
...
+}
+
+const Filter = ({selected, names, onChange}) => (
+  <DropDown
+    openLink={<a className='header-action'>{names[selected]}<Icon name='chevron-down'/></a>}
+    position='right'
+  >
+    {Object.keys(names).map(key => {
+      return(
+        <CardListElement header={names[key]} selected={names[selected] === key} onMouseDown={() => {onChange(key)}} key={key}/>
```
Comment (mmcdermott) [inline src/components/spaces/index/index.js]: Add closeOnClick=true

File: src/components/spaces/index/index.js [hunk 5]
```diff
@@ -16,29 +55,65 @@ function mapStateToProps(state) {
...
-  componentWillMount(){
-    this.props.dispatch(search.fetch('', {}))
-  }
-  _search(e) {
-    this.props.dispatch(search.fetch(e.target.value, {}))
+
+  state = {
+    sortBy: SORT_BY_POPULAR,
+    timeframe: TIMEFRAME_MONTHLY,
+    searchValue: ''
```
Comment (mmcdermott) [inline src/components/spaces/index/index.js]: Fix indentation.

File: src/components/spaces/index/index.js (hunk 1 above)
Comment (mmcdermott) [inline src/components/spaces/index/index.js]: Eliminate extra lines

File: src/components/spaces/show/index.js [hunk 6]
```diff
@@ -183,6 +183,8 @@ export default class SpacesShow extends Component {
     const owner = hasOrg ? space.organization : space.user
     const ownerUrl = hasOrg ? e.organization.url(space.organization) : e.user.url(space.user)
 
+    const canBePrivate = hasOrg ? e.organization.canMakeMorePrivateModels(space.organization) : e.me.canMakeMorePrivateModels(this.props.me)
```
Comment (mmcdermott) [inline src/components/spaces/show/index.js]: Why is this change in this branch?

File: src/components/utility/card/index.js [hunk 7]
```diff
@@ -45,30 +47,40 @@ export class CardListElement extends Component {
...
           </div>
         </a>
       </li>
     )
   }
 }
 
+const IconSection = ({colCount, icon, ionicIcon, image, imageShape}) => (
+  <div className={`col-xs-${colCount} icons`}>
+    {icon &&
```
Comment (mmcdermott) [inline src/components/utility/card/index.js]: I would one line these, but that's just a personal preference.

File: src/modules/search_spaces/actions.js [hunk 8]
```diff
@@ -7,19 +7,24 @@ import * as userActions from 'gModules/users/actions'
 export function fetch(query = '', options = {}) {
   let filters = {hitsPerPage: 21}
   filters.page = options.page || 0
-  if (options.user_id) {
-    filters.numericFilters = `user_id=${options.user_id}`
-  } else {
-    filters.numericFilters = `metric_count>2,is_private=0`
+  const {sortBy} = options
+
+  const time = new Date()
```
Comment (mmcdermott) [inline src/modules/search_spaces/actions.js]: Just do `const time = (new Date()).getTime()`

**Comment by mmcdermott** [comment]
After comments addressed, this LGTM.

**Pull request merged**